app/controllers/expenditures_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ExpendituresController < ApplicationController
+  def index
+    @expenditures = Expenditure.all.order(created_at: :desc)
+  end
+
+  def show
+    @expenditure = Expenditure.find(params[:id])
+  end
+
+  def new
+    @expenditure = Expenditure.new
+    @category = Category.find(params[:category_id])
+  end
+
+  def create
+    @category = Category.find(params[:category_id])
+    @expenditure = Expenditure.new(expenditure_params)
+    @expenditure.author_id = current_user.id
+    if @expenditure.save
+      category_expenditure = CategoryExpenditure.new(category_id: params[:expenditure][:category_id], expenditure_id: @expenditure.id)
+      category_expenditure.save
+      redirect_to category_path(@category.id)
+      flash[:notice] = 'Transaction added'
+    else
+      render :new
+      flash[:notice] = 'Error adding Transaction'
+    end
+  end
+
+  def destroy
+    @category = Category.find(params[:category_id])
+    @expenditure = Expenditure.find(params[:id])
+    @expenditure.destroy
+    flash[:notice] = 'Expenditure deleted successfully'
+    redirect_to category_path(@category.id)
+  end
```

Comment:
- [ ] It will be great to ensure that you create success and failure paths in your destroy method. What happens if the method cannot destroy the expenditure? It will be great to ensure that you have that fall back too. Just like you have done on your create method above

Reply:
Thank you for all of the feedback you make. It is supportive and I did not see while am developing. I will keep fixing it soon